WIP: Create environment variables-

haitzlm opened the pull request:
working on creating variables issue. fixes issue  #137

haitzlm commented:
I have one error to fix in the generated email:
![error](https://github.com/uclibs/staff-directory-23/assets/493654/254176c0-1391-4505-82dd-b0c193ddc265)

haitzlm commented:
image issue is now fixed as well.

Janell-Huyck commented on config/deploy/qa.rb:
```diff
@@ -2,7 +2,7 @@
 set :rails_env, :production
 set :bundle_without, %w[development test].join(' ')
-set :branch, 'qa'
+set :branch, 'createvariables'
```
You don't want to push this to qa.

Janell-Huyck replied:
It is incorrect to change the branch name in qa.  QA's branch name should remain "qa".  When you change it to do test deploys, you only change the branch name LOCALLY, and do not commit that change.

haitzlm replied:
going to get rid of this branch

Janell-Huyck commented on app/mailers/application_mailer.rb:
```diff
@@ -1,6 +1,6 @@
 
 class ApplicationMailer < ActionMailer::Base
-  default from: 'lisa.haitz@uc.edu'
+  default from: 'uclappdev@uc.edu'
```
I think you set up a default from address in your ENV variables.  You can definitely use that here.  Something like:
class ApplicationMailer < ActionMailer::Base
default_sender = ENV['MY_DEFAULT_SENDER'] || 'uclappdev@uc.edu'
default from: default_sender

Adding in the || symbol will make your code more robust in case something goes wrong with the ENV file or someone forgets to add the variable.  Since this isn't technically a secret, it's not "wrong" to hardcode like you did, but putting in an ENV call here will allow for easier changes in the future.

Janell-Huyck commented on config/initializers/devise.rb:
```diff
@@ -24,7 +24,7 @@
   # note that it will be overwritten if you use your own mailer class
   # with default "from" parameter.
-  config.mailer_sender = 'lisa.haitz@uc.edu'
+  config.mailer_sender = ENV['STADIR_PRODUCTION_MAILER_FROM']
```
👍

Janell-Huyck reviewed (commented):
The only thing that HAS to be changed is not pushing the changed deploy branch name.  That needs to stay 'qa'.  Consider the rest suggestions.  Let me know when you've got the branch name fixed and are ready for a second look.

Also, don't forget (if you haven't done this already) you'll want to put your ENV file in libappstest and in the config vault and make a note in your checklist for production deploy to add the production env file to the libapps server.  👍

Janell-Huyck commented on app/views/devise/mailer/reset_password_instructions.html.erb:
This same change shows up in your UI PR.  Be cautious with potential accidental changes/merge conflicts here.

haitzlm commented:
I changed the way i was doing the variables. Before I had this: STADIR_DEV_MAILER_FROM=uclappdev@uc.edu OR
STADIR_PRODUCTION_MAILER_URL=libappstest.libraries.uc.edu.

I changed these in favor of this format: STADIR_MAILER_URL with no production or dev indicated. This should be correct now.

Janell-Huyck commented on .env.development:
```diff
@@ -8,5 +8,7 @@ STADIR_DATABASE_PORT=
+STADIR_MAILER_FROM=lisa.haitz@uc.edu
+STADIR_MAILER_URL=libappstest.libraries.uc.edu
+STADIR_MAIL_SMTP_ADDRESS=libappstest.libraries.uc.edu
+}
```
Typo?  "}" looks out of place here.

Janell-Huyck reviewed (changes requested):
Please see 2 small things to fix and then we should be good to go

haitzlm commented:
> Please see 2 small things to fix and then we should be good to go

I got the bracket as well as set deploy/qa.rb back to qa

haitzlm commented:
i can create a user on this branch. "User.create(email: 'testuser3@gmail.com', password: 'secure123', password_confirmation: 'secure123')"

Janell-Huyck commented:
![Screenshot 2024-04-03 at 5 48 05 PM](https://github.com/uclibs/staff-directory-23/assets/57018862/4f1fc217-b460-47dd-8956-c3d28c5096c7)
And I cannot.  There's something strange going on here. :(

haitzlm commented:
You can't create the same user that I did, it will not let you.  Also, you cant use curly quotes, has to be straight. 

this branch is pulled back so i can redo it.
